pkgs/data/icons/comixcursors/default.nix

```
@@ -0,0 +1,135 @@
+
+let
+  inherit (lib.attrsets) cartesianProductOfSets;
+  inherit (lib.lists) remove;
+in
+
+let
+  owner = "limitland";
+  repo = "comixcursors";
+
+  dimensions = {
+    color = [ "Black" "Blue" "Green" "Orange" "Red" "White" ];
+    opacity = [ "" "Opaque_" ];  # Translucent or opaque.
+    thickness = [ "" "Slim_" ];  # Thick or slim edges.
+    handedness = [ "" "LH_" ];   # Right- or left-handed.
+  };
+  product =
+    cartesianProductOfSets dimensions;
+  variantName =
+    { color, opacity, thickness, handedness }:
+    "${handedness}${opacity}${thickness}${color}";
+  variants =
+    # (The order of this list is already good looking enough to show in the
+    # meta.longDescription.)
+    map variantName product;
+in
+stdenv.mkDerivation rec {
+  pname = "comixcursors";
+  version = "0.9.2";
+
+  src = fetchFromGitLab {
+    inherit owner repo;
+    # It looks like the owner forgot to make a tag for the 0.9.2 release (but
+    # previous releases have tags), so we must use the release's commit object
+    # identifier instead.
+    rev = "8c327c8514ab3a352583605c1ddcb7eb3d1d302b";
+    sha256 = "0bpxqw4izj7m0zb9lnxnmsjicfw60ppkdyv5nwrrz4x865wb296a";
+  };
+
+  nativeBuildInputs = [ bc librsvg xcursorgen ];
+
+  patches = [ ./makefile-shell-var.patch ];
+
+  postPatch = ''
+    patchShebangs .
+  '';
+
+  # install-all is used instead of the directions in upstream's INSTALL file,
+  # because using its Makefile directly is broken.  Upstream itself seems to use
+  # its build-distribution script instead, which also uses install-all, but we
+  # do not use it because it does extra things for other distros.
+  #
+  # When not all of the variants, i.e. only a smaller subset of them, are
+  # desired (i.e. when a subset of outputs are chosen), install-all will still
+  # build all of them.  While upstream appears to provide old functionality for
+  # building only a subset, it is broken and we do not use it.  With prebuilt
+  # substitutions, installers of this package will get only the outputs they
+  # chose.
+  buildPhase = ''
+    ICONSDIR=$TMP/staged ./install-all
+  '';
+
+  installPhase = ''
+    for outputName in $outputs ; do
+        if [ $outputName != out ]; then
+            local outputDir=''${!outputName};
+            local iconsDir=$outputDir/share/icons
+            local cursorName=$(tr _ - <<<"$outputName")
+
+            mkdir -p $iconsDir
+            cp -r -d $TMP/staged/ComixCursors-$cursorName $iconsDir
+
+            unset outputDir iconsDir cursorName
+        fi
+    done
+
+    # Need this directory (empty) to prevent the builder scripts from breaking.
+    mkdir -p $out
+  '';
+
+  outputs = let
+    default = "Opaque_Black";
+  in
+    # Have the most-traditional variant be the default output (as the first).
+    # Even with outputsToInstall=[], the default/first still has an effect on
+    # some Nix tools (e.g. nix-build).
+    [ default ] ++ (remove default variants)
```

Why do we need to do this dance? Can't we just use variants as is?


----


```suggestion
    [ default ] ++ (lib.remove default variants)
```

---

My reasons for this dance are:

1. It enables potentially changing the default/first output in the future by only changing the value of the `default` variable, which I thought is good for maintenance, without needing to reorder the sub-values of the `dimensions` variable (which determine the order of the `variants` value).

2. I thought the first output, sometimes used as default, should be `"Opaque_Black"`, but the first element of `variants` is `"Black"` which is translucent.  If `variants` was just used as-is, with the current order of `dimensions` sub-values, then the first would not be what I intended; and, also, I intended `dimensions` to have its current order because I thought that is the most logical and readable.

I'd be fine with changing these aspects however you want - maybe change the order of only `dimensions.opacity` so that `"Opaque_Black"` is first?  Or we could accept translucent `"Black"` as the first/default (maybe the creator of Comix Cursors intended translucency to be the default, given their naming scheme)?

(Note that the first output as default is only used by some things like `nix-build` but is not used by things like `nixos-rebuild` because I made this package have `outputsToInstall = []`.)

Thanks for the review!

---

Oh, then we can keep it.